ui-modules/home/app/views/main/deploy/deploy.controller.js

```diff
@@ -80,9 +79,17 @@ export function deployStateController($scope, $state, $stateParams, $uibModal, b
 
     function modalController($scope, $location, entitySpec, locations) {
         $scope.app = entitySpec;
-        $scope.locations = locations;
-        // can optionally add: { noEditButton: true, noComposerButton: true }, or pass in URL
-        $scope.args = angular.extend({}, $location.search());
+        $scope.locations = filterCatalogQuickLaunch(locations, (t) => {
+                $scope.usingCatalogQuickLaunchTags = t.length > 0;
+            });
+        
+        // also supports { noEditButton: true, noComposerButton: true }
+        // see quick-launch.js for more info
+        $scope.args = angular.extend({
+                noCreateLocationLink: $scope.usingCatalogQuickLaunchTags
```

Comment:
Why disabling the location creation button if there are quick launch items? It doesn't really make sense to me.

Reply:
the thinking is:

```
+Additionally if any locations have that tag, the "Create Location" option is suppressed,
+as it is assumed the Quick Launch is being used in a curated environment where that option is not meant for end-users.
+(This can be changed through config options in the code, for users who wish to produce their own UI.)
```

which i've added in https://github.com/apache/brooklyn-docs/pull/299/commits/4df37efc19047723ca24e559e398185a4ba08455 (just now).  also changed the var name to make it a little clearer -- `usingLocationCatalogQuickLaunchTags`.  (ie to be clear, this only applies if a _location_ is tagged, not if apps are tagged.)

Reply:
Ah got it. Would be good to make it explicitly configurable but we can live with it for now.